src/utils/self_test/self_test.c

```diff
@@ -1674,119 +1918,192 @@ int main(int argc, char *argv[])
 	int				 c;
 	int				 j;
 	int				 ret = 0;
-	struct st_endpoint		*endpts = NULL;
-	struct st_endpoint		*ms_endpts = NULL;
-	uint32_t			 num_endpts = 0;
-	uint32_t			 num_ms_endpts = 0;
-	int				 output_megabits = 0;
-	int16_t				 buf_alignment =
-		CRT_ST_BUF_ALIGN_DEFAULT;
-	char				*attach_info_path = NULL;
+	int				 dump = 0;
+
+	static struct option long_options[] = {
+		{"file-name", required_argument, 0, 'f'},
+		{"config", required_argument, 0, 'c'},
+		{"display", no_argument, 0, 'd'},
+
+		{"group-name", required_argument, 0, 'g'},
+		{"master-endpoint", required_argument, 0, 'm'},
+		{"endpoint", required_argument, 0, 'e'},
+		{"message-sizes", required_argument, 0, 's'},
+		{"repetitions-per-size", required_argument, 0, 'r'},
+		{"max-inflight-rpcs", required_argument, 0, 'i'},
+		{"align", required_argument, 0, 'a'},
+		{"Mbits", no_argument, 0, 'b'},
+		{"singleton", no_argument, 0, 't'},
+		{"randomize-endpoints", no_argument, 0, 'q'},
+		{"path", required_argument, 0, 'p'},
+		{"nopmix", no_argument, 0, 'n'},
+		{"help", no_argument, 0, 'h'},
+		{0, 0, 0, 0}
+	};
+	extern int			optind;
+
+	g_msg_sizes_str = default_msg_sizes_str;
+	g_rep_count = g_default_rep_count;
+	g_max_inflight = g_default_max_inflight;
 
 	ret = d_log_init();
 	if (ret != 0) {
 		fprintf(stderr, "crt_log_init() failed. rc: %d\n", ret);
 		return ret;
 	}
-	/********************* Parse user arguments *********************/
+
+	/****************** First Parse user file arguments *************/
+	/* File specified via -f file argument */
 	while (1) {
-		static struct option long_options[] = {
-			{"group-name", required_argument, 0, 'g'},
-			{"master-endpoint", required_argument, 0, 'm'},
-			{"endpoint", required_argument, 0, 'e'},
-			{"message-sizes", required_argument, 0, 's'},
-			{"repetitions-per-size", required_argument, 0, 'r'},
-			{"max-inflight-rpcs", required_argument, 0, 'i'},
-			{"align", required_argument, 0, 'a'},
-			{"Mbits", no_argument, 0, 'b'},
-			{"singleton", no_argument, 0, 't'},
-			{"randomize-endpoints", no_argument, 0, 'q'},
-			{"path", required_argument, 0, 'p'},
-			{"nopmix", no_argument, 0, 'n'},
-			{0, 0, 0, 0}
-		};
+		c = getopt_long(argc, argv, "hf:c:dg:m:e:s:r:i:a:btnqp:",
+				long_options, NULL);
+		/* break out of while loop */
+		if (c == -1)
+			break;
+
+		switch (c) {
+		case 'f':
+			printf(" file name %s\n", optarg);
+			file_name = optarg;
+			break;
+		case 'c':
+			section_name = optarg;
+			break;
+		case 'd':
+			dump = 1;
+			break;
+		case 'h':
+			print_usage(argv[0], default_msg_sizes_str,
+				    g_default_rep_count,
+				    g_default_max_inflight);
+			D_GOTO(cleanup, ret = -DER_INVAL);
+		default:
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment

Reply:
fixed